isis/src/system/objs/KernelDb/KernelDb.cpp

```diff
@@ -581,20 +582,34 @@ namespace Isis {
     bool matchTime = !grp.hasKeyword("Time");
     bool matchKeywords = true;
 
+    // If the kernel segment has an instrument specification that doesn't match
+    // the instrument id in the label, don't perform the rest of the checks
+    if (grp.hasKeyword("Instrument")){
+        const PvlGroup &inst = lab.findGroup("Instrument", Pvl::Traverse);
+        if (inst["InstrumentId"][0].toStdString() != grp["Instrument"][0].toStdString()){
+          return false;
+        };
+    }
+
```

Comment:
I'm not sure we want to fall out of the checks completely if this is the case. Say we have db with both Themis VIS and Themis IR kernel selections. If one of the selection groups has an instrument then Themis VIS would fail the matching. I believe that should instead it should try to match with the other selections that do not have an instrument listed before failing the matching completely.

Reply:
Since only selection groups with offsets have this instrument keyword, it might be better to do this check after trying to apply the offset? I am not sure.